### ISSUE TITLE
Uncaught error: packager is not running at localhost: 19001

### ISSUE THREAD

**Issue by nahidmbstu** [expo/expo#1381]
## Description
after giving the command exp start
dependency graph loaded....
i scan my QR code ....
i view the error log ...
Uncaught error: packager is not running at localhost: 19001
i have a open console http:// localhost: 19001/deggger-ui/
app does not open

**Comment by terribleben** [comment]
Hello! Sorry you're having a trouble. Here are a few things you can try:

- Check for any other messages in the error log before that one, which may have prevented the packager from starting
- If you're running an Android device, make sure your device can open localhost:19001 in the device's browser. (this would not be possible from an iOS device)
- Consider [clearing your packager cache](https://slack-redir.net/link?url=https%3A%2F%2Fforums.expo.io%2Ft%2Fhow-to-clear-the-react-native-packager%2F1352) if nothing appears to be working.

If you're still stuck, please open another issue and follow the issue template (including providing us some info about your environment.)

**Comment by kanteankit** [comment]
Hi I am also facing this issue. I am using CRNA and yarn.
Before this problem, I was using `yarn start` to start the packager. The URL it showed was `exp://192.168.1.15:19001`.
Then my wifi went off.So in order to run my app using a USB cable,I ran the following command: `exp start --localhost --android`. This enabled me to run my CRNA app using USB cable.The URL it showed was: `exp://localhost:19000`

But,when my wifi came back and I tried to start the packager using `yarn start`, then this error message started showing up. Now, I cannot run my app over wifi.
Please help, I cannot find a solution for this.

**Comment by Gaboik** [comment]
I found that this error message could arise for a very simple reason, make sure your cellphone running Expo is connected to the same sub network  as your computer on which your dev server is running.  Like @terribleben said,  

> If you're running an Android device, make sure your device can open localhost:19001 in the device's browser. (this would not be possible from an iOS device)

I realized that my cellphone was not connected to my LAN and was getting Internet connection by satellite.

**Comment by HanifCarroll** [comment]
Kind of unrelated, but perhaps this information will help someone in the future:

I ran into this error message while trying to connect my physical device.  I didn't want to use the Tunnel host because it was always slow in my experience, and took ages for changes to get passed through.  My usual method of LAN wasn't working for some reason (I hadn't used Expo in a while, and changed my OS), and I tried to use Localhost to solve the problem instead.  That's when I got the error described in the OP.  

I don't have a solution for that error, but I can share how I fixed things to make the LAN host option work again.  Turns out it was my VPN on my phone, PIA to be specific.  Having the VPN enabled on my computer is no problem, but having it enabled on my phone doesn't allow the connection to form.  Once I turned it off, things were back to normal.

**Comment by Grebe49** [comment]
I get this message and cannot get beyond it. Running on windows 10,  npm start tells me that the app is running at exp://192.169.1.158:19000 - when I scan the QR code on my android tablet, I get the message "uncaught exception: Packager is not running at http://192.168.1.158:190001". The tablet is on the same netwok at the PC, and I have tried to clear the cache and restarted NPM  (SHIFT R) to no effect.

As I write this I remembered that I had updated my firewall since last running - turning the firewewall off reolved the issue. Left the comment as it may help someone else.

**Comment by vinkomlacic** [comment]
I resolved this issue by setting the network profile to private network. It looks like it was an error caused by the firewall. After setting the network profile I got asked about whether I should allow Node.js something concerning the firewall. Hope that helps.

**Comment by bonham000** [comment]
I'm experiencing this repeatedly. I can run Expo locally through the cli or XDE and open the project in the simulator, but trying to open on the Expo app on an iOS or Android device (on the same wifi as my computer) results in errors about the packager not running: `Something went wrong. Packager is not running at http://{{local-ip}}:19001` or on iOS `There was a problem running the requested app.`

If I tether my computer to my Android phone, and run the packager that way, my phone can connect to the development server.

I don't have any Firewall or VPN settings enabled.

Also, worth mentioning this feature was working fine for me earlier but has recently stopped working. I have upgraded to Expo 29.

**Comment by ianemv** [comment]
Also experiencing this issue after my installation got updated to Expo 29 and installing react navigation.

**Comment by IshanUdyoga** [comment]
I'm experiencing this issue .. After changing local IP address works fine :-)

**Comment by mlapeter** [comment]
Just wanted to add for anyone else googling this: for me I get this error randomly, my guess is due to spotty wifi in my house. At first I tore my hair out trying all sorts of solutions, but currently I find if I just try `exp start` again 2-3 times it then works for me. That, and turning wifi off and on for both phone and laptop helps.

**Comment by adamtaylor13** [comment]
@mlapeter Your suggestion of turning off the WiFi on both devices worked like a charm. 👌🏻

**Comment by badarshahzad** [comment]
@mlapeter  your advice work for me also.

**Comment by angyy10** [comment]
my issue was my phone is connecting to 2.4ghz wifi while my mac is connecting to 5ghz wifi, had to switch my mac to same wifi network with my phone

**Comment by theknightsky** [comment]
I was able to solve this problem simply by opening up the specified URL in my browser, and then rerunning
``` expo start ```

**Comment by kmui2** [comment]
Found that my laptop was connect via ethernet and solved this by connecting to the wifi instead.

**Comment by devmuhammad** [comment]
Had the same issue, Apparently, my bundler was running twice..had to delete on of the instances. Works now !

**Comment by Daniyal-Najeeb** [comment]
Disable any other adapter running besides the your  LAN network adapter restart your expo server, maybe this trick can resolve your issue. because I had the same issue and I used this procedure to resolve my issue.

**Comment by inspireui** [comment]
Delete the .expo folder and run expo start again works for me

**Comment by jleonelpm** [comment]
Thanks @inspireui worked for me, too.

**Comment by AJamesPhillips** [comment]
Stopped and started my app running through Expo app on phone - problem not solved.
Stopped and started Expo app itself - problem solved.

**Comment by Dave144** [comment]
I just switched to WIFI from  mobile internet and it went OK! So simple.try it

**Comment by AhmedMKamal** [comment]
I went through the same issue today on an Ubuntu machine and it was the firewall.

**Comment by rosnaib11** [comment]
@AhmedMKamal can you show what you did please

**Comment by AhmedMKamal** [comment]
@rosnaib11 you can disable it by running `sudo ufw disable` or just enable the specific port by running `sudo ufw allow 19001` keep in mind this going to work only if it's a firewall issue as the case with me.

**Comment by ivosh** [comment]
@ghost Your recipe to delete `.expo` folder and start `expo` again worked for me. Thanks, man!

**Comment by Sunrays** [comment]
I'm running an expo ejected app and developing with ExpoKit when I got this problem. Using Android mobile as emulator. The app didn't load through LAN or Local and loads only through Tunnel which is slow so not ideal for development. I tried many ways to fix the problem including
- Deleting `.expo` folder.
- Restarting expo multiple times.
- Exiting Terminal.
- Switching off WiFi in laptop as well as mobile.
At last after clearing cache on expo app on mobile, uninstalling my app and building again by `./gradlew installDevKernelDebug` in Expo CLI I could able to get it working.

**Comment by bondansebastian** [comment]
After 2 hours of frustration, resetting my router solved the problem.

**Comment by atik-trillit** [comment]
Getting same issue. Restarting Wifi of system(switch off and then on) solve the problem.

**Comment by DavidKHahn** [comment]
Hello Guys, 

I am having a similar problem except my app works perfectly fine while on Wifi however when it is not on wifi I can't get it to run. Any leads would be appreciated!

**Comment by ilivestrong** [comment]
This thread might be old, but just to add my bit.
The problem could be, as was in my case, that development machine/laptop as well as your Mobile device should be on same network. This would be the case if you selected "LAN" on the Expo Devtools and scanned the displayed QR Code.

Good luck!

**Comment by kushal9291** [comment]
I had the same issue. What works for me was rebooting my router.

**Comment by Batsirai-sudo** [comment]
> I would like to share my experience on this problem. I am using WSL on windows 10 and I tried all methods mentioned above with no luck. 
> Finally I discovered that port 19001(used by the bundler) was blocked by windows defender firewall( interestingly port 19000 is opened already).
> If you guys are facing the same issue on windows 10 you can go to windows defender firewall settings and add an inbound rule which open port 19001.

this helped alot

**Comment by JonatanOrtiz** [comment]
> Found that my laptop was connect via ethernet and solved this by connecting to the wifi instead.

@kmui2  worked like a charm!

**Comment by tango07** [comment]
I connected my mobile and laptop on the same connection i.e., I turned on the mobile hotspot and connected with the laptop and everything worked!

**Comment by Thomazella** [comment]
turned on wifi on phone, it works now

**Comment by sulaimon-olaniran** [comment]
> Delete the .expo folder and run expo start again works for me

worked for me after trying several other options,
